cineast-api/src/main/java/org/vitrivr/cineast/api/messages/lookup/BooleanLookup.java

```diff
@@ -0,0 +1,139 @@
+package org.vitrivr.cineast.api.messages.lookup;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.apache.commons.lang3.tuple.MutableTriple;
+import org.apache.commons.lang3.tuple.Triple;
+import org.graalvm.compiler.nodes.calc.ObjectEqualsNode;
+import org.vitrivr.cineast.api.messages.interfaces.Message;
+import org.vitrivr.cineast.api.messages.interfaces.MessageType;
+import org.vitrivr.cineast.core.data.providers.primitive.PrimitiveTypeProvider;
+import org.vitrivr.cineast.core.db.RelationalOperator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BooleanLookup implements Message {
+
+
+    /**
+     * The type of Boolean Lookup
+     */
+
+    private String type;
+
+    /**
+     * Entity for the BooleanQuery
+     */
+    private String entity;
+    /**
+     * List consisting of single Boolean Queries
+     */
+    private List<BooleanLookupQuery> queries;
+    /**
+     * QueryList to be used in the Boolean Feature
+     */
+    private List<Triple<String, RelationalOperator, List<PrimitiveTypeProvider>>> queryList;
+    /**
+     * ComponentID to map a query to a BoolTerm in the frontend
+     */
+    private int componentID;
+
+    /**
+     * Constructor for the BooleanLookup object.
+     *
+     */
+
+    @JsonCreator
+    public BooleanLookup(@JsonProperty("boolQueries") List<BooleanLookupQuery> queries, @JsonProperty("type") String type,
+                         @JsonProperty("componentID") int componentID) {
+        this.queries = queries;
+        this.type = type;
+        this.componentID = componentID;
+        this.entity = queries.get(0).getEntity();
+        this.queryList = new ArrayList<Triple<String, RelationalOperator, List<PrimitiveTypeProvider>>>();
+        createTriple();
+    }
+
+    private void createTriple() {
+        for (BooleanLookupQuery query: queries) {
+            queryList.add(new MutableTriple<>(query.getAttribute(), query.getOperator(), query.getValue()));
+        }
+}
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public MessageType getMessageType() {
+        return MessageType.B_LOOKUP;
+    }
+
+    @Override
+    public String toString() {
+        return ReflectionToStringBuilder.toString(this, ToStringStyle.JSON_STYLE);
+    }
+
+    public List<BooleanLookupQuery> getQueries() { return queries; }
+
+    public String getType() { return type; }
+
+    public String getEntity() {
+        return entity;
+    }
+
+    public int getComponentID() { return componentID; }
+
+    public List<Triple<String, RelationalOperator, List<PrimitiveTypeProvider>>> getQueryList() { return queryList; }
+
+}
+class BooleanLookupQuery {
```

Comment:
Please put this into a different file, we're trying to only have one class per file